src/firebaseConfig.ts

```diff
@@ -5,74 +5,146 @@
 // 'npm run generate:json-schema' to regenerate the schema files.
 //
 
-export type DatabaseConfig =
+// Sourced from - https://firebase.google.com/docs/hosting/full-config#rewrite-cloud-run-container
+type CloudRunRegions =
+  | "asia-east1"
+  | "asia-east2"
+  | "asia-northeast1"
+  | "asia-northeast2"
+  | "asia-northeast3"
+  | "asia-south1"
+  | "asia-southeast1"
+  | "asia-southeast2"
+  | "australia-southeast1"
+  | "europe-north1"
+  | "europe-west1"
+  | "europe-west2"
+  | "europe-west3"
+  | "europe-west4"
+  | "europe-west6"
+  | "northamerica-northeast1"
+  | "southamerica-east1"
+  | "us-central1"
+  | "us-east1"
+  | "us-east4"
+  | "us-west1";
+
+type CloudFunctionRuntimes = "nodejs10" | "nodejs12" | "nodejs14";
+
+type DeployAsset = {
+  predeploy?: string | string[];
+  postdeploy?: string | string[];
+};
+
+type DatabaseOne = {
+  rules: string;
+} & DeployAsset;
+
+type DatabaseMany = ({
+  rules: string;
+} & (
   | {
-      rules?: string;
+      instance: string;
```

Comment:
This took me a while to read, maybe just inline the `rules` field? So:

```
{
      target?: string;
      instance: string;
      rules: string;
 } | {
      target: string;
      instance?: string;
      rules: string;
    }
```

Reply:
I used the `RequireAtLeastOne` type mentioned below.